File: frontend/src/app/models/Ticket.ts
export enum TicketStatus {
  OPEN,
  ASSIGNED,
  IN_PROGRESS,
  CLOSED,
}

export interface Ticket {
  id: number;
  title: string;
  description: string;
  status: TicketStatus;
  createdAt: Date;
  updatedAt: Date;
}

export type CreateTicket = Pick<Ticket, 'title' | 'description'>;

File: backend/src/main/java/dev/tizmr/OpenTicket/domain/entity/Ticket.java
package dev.tizmr.OpenTicket.domain.entity;

import jakarta.persistence.*;
import java.time.Instant;
import java.util.UUID;

import lombok.*;

@Entity
@Table(name = "tickets")
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Ticket {

  @Id
  @GeneratedValue(strategy = GenerationType.UUID)
  @Column(updatable = false, nullable = false)
  private UUID id;

  @Column(nullable = false)
  private String title;

  @Column(nullable = false)
  private String description;

  @Enumerated(EnumType.STRING)
  @Column(nullable = false)
  private TicketStatus status;

  @Column(name = "created_at", nullable = false, updatable = false)
  private Instant createdAt;

  @Column(name = "updated_at", nullable = false)
  private Instant updatedAt;
}
